src/NodeAutoCompleteViewExtension/ViewModels/NodeAutoCompleteBarViewModel.cs

```diff
@@ -795,14 +823,22 @@ internal void AddCluster(int filterredIndex)
             var currentFilter = FilteredView.Cast<DNADropdownViewModel>().ToList();
             var currentItem = filterredIndex >= 0 && filterredIndex < currentFilter.Count ? currentFilter[filterredIndex] : null;
 
-            AddCluster(currentItem.ClusterResultItem);
+            if (currentItem?.ClusterResultItem != null)
+            {
+                AddCluster(currentItem.ClusterResultItem);
+            }
         }
 
         // Add Cluster from server result into the workspace
         internal void AddCluster(ClusterResultItem clusterResultItem)
         {
             if (clusterResultItem == null || clusterResultItem.Topology == null)
                 return;
+            if (lastAddedCluster == clusterResultItem)
```

Comment:
Maybe we should add an id or implement Equals for the clusterresultitem. The reference comp might get back at us in the future.

Reply:
changed to a comparison by json serializer